optional/slides/slides.md

```diff
@@ -0,0 +1,207 @@
+class: center, middle
+
+# Optional
+
+## A better alternative to null
+Raise your hand if you ever got a NullPointerException during your life as a Java developer.
+
+---
+
+# The inventor
+
+.center[![Billion-Dollar](http://www.azquotes.com/picture-quotes/quote-i-call-it-my-billion-dollar-mistake-it-was-the-invention-of-the-null-reference-in-1965-tony-hoare-113-96-46.jpg)]
+
+---
+
+# How do you model the absence of a value?
+
+```
+public class Person {
+	private Car car;
+	public Car getCar() { return car; }
+}
+public class Car {
+	private Insurance insurance;
+	public Insurance getInsurance() { return insurance; }
+}
+public class Insurance {
+	private String name;
+	public String getName() { return name; }
+}
+```
+---
+
+### Then, what’s possibly problematic with the following code?
+
+```
+public String getCarInsuranceName(Person person) {
+	return person.getCar().getInsurance().getName();
+}
+```
+
+_Will it blend?_
+
+```
+public String getCarInsuranceName(Person person) {
+	if (person != null && person.getCar() != null &&
+			person.getCar().getInsurance() != null)
+		return person.getCar().getInsurance().getName();
+	else
+		return "Unknow Name";
+}
+```
+
+---
+
+# So what?
+
+* **A source of error.** NullPointerException is by far the most common exception in Java.
+* **It bloats your code.** Deeply nested null checks.
+* **It’s meaningless.** It doesn’t have any semantic meaning.
+* **It breaks Java philosophy.** Java always hides pointers from developers except in one case: _the null pointer_.
+* **It creates a hole in the type system.** null carries no type or other information, meaning it can be assigned to any reference type.
+
+_And in particular it represents the wrong way to model the absence of a value in a statically typed language._
+
+---
+
+## Introducing the Optional class
+
+Java 8 introduces a new class called `java.util.Optional<T>` that’s inspired by the ideas of Haskell and Scala.
+
+It’s a class that encapsulates an optional value.
+
+When a value is present, the Optional class just wraps it.
+Conversely, the absence of a value is modeled with an “empty” optional returned by the method `Optional.empty()`.
+
+You might wonder what the difference is between a null reference and `Optional.empty()` which is a valid, workable object of type Optional that can be invoked in useful ways.
+
+???
+Absent is a static factory method that returns a special singleton instance of the Optional class.
+
+---
+
+## Introducing the Optional class, for Java 6
+
+### Yes! Guava has `com.google.common.base.Optional<T>` too!
+
+Unlike the Java one this implements _Serializable_!
+
+Has similar methods but with different names:
+* `fromNullable` and `ofNullable`
+* `absent` and `empty`
+* `or` and `orElse`
+* `transform` and `map`/`flatMap`
+
+But sometimes not: `get`, `isPresent`.
+
+???
+This can be used like a field.
```

Comment:
not sure of this. The pattern is clear: no optional as a field, return optional if field is null (or if null represents an absence of value)